packages/@react-aria/overlays/test/useOverlayTrigger.test.js

```diff
@@ -16,7 +16,7 @@ import {useOverlayTrigger} from '../';
 
 function Example(props) {
   let ref = useRef();
-  useOverlayTrigger({ref, ...props});
+  useOverlayTrigger(props, ref);
```

Comment:
given that some are `props, state, ref`, if we want the signatures to match if we need to add state, should we use named args in an options object?

Reply:
this is a legitimate concern, and already exists in several hooks. we have 3 options:

1. Change all hooks to accept an options object instead of 3 args. Seems like a big change.
2. For hooks that accept only props and ref, add a state arg in the middle that you have to pass `null` to. 😬 
3. Do nothing. Add state at the end if we need to later, and hope we don't.

Personally, I prefer (3) for most of the components where it's very unlikely we'd add state (e.g. button). That said, I think this brings up a good point for `useOverlayTrigger` specifically. I think we should be passing the state in to it rather than the current "props". Will handle in a separate PR.

Reply:
i'm uneasy about 3, while 1 is a sweeping change, i don't think it's likely to cause trouble, and it would quickly be apparent if we missed refactoring one of them. 3 just sounds like a setup for potential divergence of these signatures. for instance, if we build the transition stuff into button like react suggests, then button may have state